parlai/crowdsourcing/utils/tests.py

```diff
@@ -408,3 +408,26 @@ def _send_agent_message(
             "episode_done": False,
         }
         self.server.send_agent_act(agent_id=agent_id, act_content=act_content)
+
+
+def check_stdout(actual_stdout: str, expected_stdout_path: str):
+    """
+    Check that actual and expected stdouts match.
+
+    Given a string of the actual stdout and a path to the expected stdout, check that
+    both stdouts match, keeping in mind that the actual stdout may have additional
+    strings relating to progress updates that are not found in the expected output
+    strings.
+
+    TODO: this can probably be moved to a method of an abstract test class once all
+     analysis code relies on pytest regressions for some of its tests.
+    """
+    actual_stdout_lines = actual_stdout.split('\n')
+    with open(expected_stdout_path) as f:
+        expected_stdout = f.read()
+    for expected_line in expected_stdout.split('\n'):
```

Comment:
in your tests, have you found this op to take a while? it seems inefficient complexity-wise but I don't have a suggestion for anything better

Reply:
Good point about the inefficiency - the desired output is only a few dozen lines, and in practice the quicker of the two tests that uses this takes roughly a quarter of a second total to complete